src/main/java/org/traccar/protocol/EgtsProtocolDecoder.java

```diff
@@ -254,8 +274,17 @@ protected Object decode(
                 buf.readerIndex(end);
             }
 
-            if (serviceType == SERVICE_TELEDATA && deviceSession != null) {
-                positions.add(position);
+            if (serviceType == SERVICE_TELEDATA && position.getFixTime() != null) {
```

Comment:
Why are we checking `position.getFixTime() != null` here?

Reply:
position is created before loop, and when loop ends position is empty some times. Should we change this to position.isValid()?